overlord/snapstate/snapstate.go

```diff
@@ -1070,6 +1076,88 @@ func FinishTaskWithRestart(task *state.Task, status state.Status, rt restart.Res
 	return restart.FinishTaskWithRestart(task, status, rt, rebootRequiredSnap, rebootInfo)
 }
 
+func isChangeRequestingSnapdRestart(chg *state.Change) bool {
+	// during refresh of the snapd snap, after the services of new snapd
+	// have been set up in link-snap, daemon restart is requested, link-snap
+	// is marked as Done, and the auto-connect task is held off (in Do or
+	// Doing states) until the restart completes
+	var haveSnapd, linkDone, autoConnectWaiting bool
+	for _, tsk := range chg.Tasks() {
+		kind := tsk.Kind()
+		switch kind {
+		case "link-snap", "auto-connect":
+			// we're only interested in link-snap and auto-connect
+		default:
+			continue
+		}
+
+		tss, err := TaskSnapSetup(tsk)
```

Comment:
nitpick: this is an unusual name for a SnapSetup, we always call it snapsup and a tss is usually a taskset
```suggestion
		snapsup, err := TaskSnapSetup(tsk)
```